remote/mdb_service_scripting.go

```diff
@@ -18,11 +23,198 @@ const (
 	ScriptingTestCommand      = "test_scripting"
 )
 
-func (s *mdbService) scriptingGet(ctx context.Context, w io.Writer, msg mongowire.Message)       {}
-func (s *mdbService) scriptingCreate(ctx context.Context, w io.Writer, msg mongowire.Message)    {}
-func (s *mdbService) scriptingSetup(ctx context.Context, w io.Writer, msg mongowire.Message)     {}
-func (s *mdbService) scriptingCleanup(ctx context.Context, w io.Writer, msg mongowire.Message)   {}
-func (s *mdbService) scriptingRun(ctx context.Context, w io.Writer, msg mongowire.Message)       {}
-func (s *mdbService) scriptingRunScript(ctx context.Context, w io.Writer, msg mongowire.Message) {}
-func (s *mdbService) scriptingBuild(ctx context.Context, w io.Writer, msg mongowire.Message)     {}
-func (s *mdbService) scriptingTest(ctx context.Context, w io.Writer, msg mongowire.Message)      {}
+func (s *mdbService) scriptingGet(ctx context.Context, w io.Writer, msg mongowire.Message) {
+	req := &scriptingGetRequest{}
+	if !s.serviceScriptingRequest(ctx, w, msg, req, ScriptingGetCommand) {
+		return
+	}
+
+	harness := s.getHarness(ctx, w, req.ID, ScriptingGetCommand)
+	if harness == nil {
+		return
+	}
+
+	s.serviceScriptingResponse(ctx, w, nil, ScriptingGetCommand)
+}
+
+func (s *mdbService) scriptingCreate(ctx context.Context, w io.Writer, msg mongowire.Message) {
+	req := &scriptingCreateRequest{}
+	if !s.serviceScriptingRequest(ctx, w, msg, req, ScriptingCreateCommand) {
+		return
+	}
+
+	opts, err := options.NewScriptingHarness(req.Params.Type)
+	if err != nil {
+		shell.WriteErrorResponse(ctx, w, mongowire.OP_REPLY, errors.Wrap(err, "problem creating harness options"), ScriptingCreateCommand)
+	}
+	if err = bson.Unmarshal(req.Params.Options, opts); err != nil {
+		shell.WriteErrorResponse(ctx, w, mongowire.OP_REPLY, errors.Wrap(err, "problem unmarshalling options"), ScriptingCreateCommand)
+	}
+
+	harness, err := s.harnessCache.Create(s.manager, opts)
+	if err != nil {
+		shell.WriteErrorResponse(ctx, w, mongowire.OP_REPLY, errors.Wrap(err, "problem creating harness"), ScriptingCreateCommand)
+		return
+	}
+
+	s.serviceScriptingResponse(ctx, w, makeScriptingCreateResponse(harness.ID()), ScriptingCreateCommand)
+}
+
+func (s *mdbService) scriptingSetup(ctx context.Context, w io.Writer, msg mongowire.Message) {
+	req := &scriptingSetupRequest{}
+	if !s.serviceScriptingRequest(ctx, w, msg, req, ScriptingSetupCommand) {
+		return
+	}
+
+	harness := s.getHarness(ctx, w, req.ID, ScriptingSetupCommand)
+	if harness == nil {
+		return
```

Comment:
I see you checked for nil harness cache a bunch of times. Given that the only entry point to make an `mdbService` is through `StartMDBService()`, is it really necessary to do lots of nil checking for whether the service has a harness when we know it will have one?

Reply:
Maybe not? But this also checks for the ID in that harness, which could not exist.

Reply:
Fair. I think it's fine to handle that the ID might not exist but the cache itself should definitely exist. Can we get rid of just the checks for whether `s.harnessCache` is `nil` since we already know it's non-nil?